torch/csrc/utils/tensor_new.cpp

```diff
@@ -148,6 +148,15 @@ std::vector<int64_t> compute_sizes(PyObject* seq) {
 }
 
 ScalarType infer_scalar_type(PyObject *obj) {
+#ifdef USE_NUMPY
+  if (PyArray_Check(obj)) {
+    return numpy_dtype_to_aten(PyArray_TYPE((PyArrayObject*)obj));
+  }
+  if (PyArray_CheckScalar(obj)) {
+    THPObjectPtr arr(PyArray_FromScalar(obj, nullptr));
+    return numpy_dtype_to_aten(PyArray_TYPE((PyArrayObject*) arr.get()));
+  }
+#endif
```

Comment:
BTW, while you're modifying this file, can you make `infer_scalar_type` static? There is another function called `infer_scalar_type` in the codebase and I think that they're distinct.

Reply:
NVM, I see that it is in an anonymous namespace, so no need to make it static.

Reply:
updated

Reply:
oops